subprojects/core/src/main/java/org/gradle/execution/FinalizeBuildOutputCleanupRegistry.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.execution;
+
+import org.gradle.internal.execution.BuildOutputCleanupRegistry;
+
+public class FinalizeBuildOutputCleanupRegistry implements BuildConfigurationAction {
```

Comment:
How about we call the other thing the `BuildOutputCleanupRegistryBuilder`, and this one the `BuildOutputCleanupRegistry`?

Reply:
So you would want to have an actual immutable thing as `BuildOutputCleanupRegistry` created by a builder? That would be nice, though we can't add services later in the stage to a given scope.

Or did you mean something else?